sdk/include/opentelemetry/sdk/common/attributemap_hash.h

```diff
@@ -74,27 +74,6 @@ inline size_t GetHashForAttributeMap(const OrderedAttributeMap &attribute_map)
   return seed;
 }
 
-// Calculate hash of keys and values of KeyValueIterable, filtered using callback.
-inline size_t GetHashForAttributeMap(
```

Comment:
It is not expected to calculate hash on `KeyValueIterable` by traversing it, because the order of the elements should not affect the hashing. `MetricAttributes` need to be constructed from it before hash. So remove this function.